### PR TITLE
direnv support

### PR DESCRIPTION
**Pull request by glyh**
### Status
Ready for merge.

### Description
Direnv is a tool for user to separate their shell config into different folder specific profile. When `cd`ing into the `oil` folder, if direnv is allowed to run the script, it will automatically run the `build/dev-shell.sh` script. and dump environment variables into current shell.

### Usage
- [Install direnv and its shell hook](https://direnv.net/#getting-started)
- cd into git repo root
- Run `direnv allow`
- Now whenever you cd into oil repo or one of its subdirs, direnv will load environment variables directly for you.

### CODE REVIEW

File: .envrc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
```
Comment (andychu) [inline .envrc:2]: Please add some comments on what this is, what the purpose is (run bin/ysh etc. without changing $PATH)

how to use it, how to install, etc.


I've actually never used direnv before :)

**Review by andychu — COMMENTED** [review]
OK thanks, I think this is a good idea

Just needs some comments

**Comment by glyh** [comment]
This is somewhat broken, I need to fix it.

File: .envrc (hunk 1 above)
Comment (glyh) [reply .envrc:2]: Okay this is done

Comment (andychu) [reply .envrc:2]: OK great, can you add the comments to the top of the script?  So people know how to use it


Right now it looks like it's only in the PR, and will get lost

Comment (glyh) [reply .envrc:2]: done

**Comment by andychu** [comment]
Great, thank you!  I will "advertise" this on Zulip

**Pull request merged**